FTBFS clang-18

```
2024/08/13 22:51:40 INFO -- Found LibXml2: /usr/lib/libxml2.so (found version "2.13.3")
2024/08/13 22:51:40 WARN CMake Error at /usr/lib/cmake/llvm/LLVMExports.cmake:1683 (message):
2024/08/13 22:51:40 WARN   The imported target "LLVMDemangle" references the file
2024/08/13 22:51:40 WARN
2024/08/13 22:51:40 WARN      "/usr/lib/libLLVMDemangle.a"
2024/08/13 22:51:40 WARN
2024/08/13 22:51:40 WARN   but this file does not exist.  Possible reasons include:
2024/08/13 22:51:40 WARN
2024/08/13 22:51:40 WARN   * The file was deleted, renamed, or moved to another location.
2024/08/13 22:51:40 WARN
2024/08/13 22:51:40 WARN   * An install or uninstall procedure did not complete successfully.
2024/08/13 22:51:40 WARN
2024/08/13 22:51:40 WARN   * The installation package was faulty and contained
2024/08/13 22:51:40 WARN
2024/08/13 22:51:40 WARN      "/usr/lib/cmake/llvm/LLVMExports.cmake"
2024/08/13 22:51:40 WARN
2024/08/13 22:51:40 WARN   but not all the files it references.
```

So https://github.com/wolfi-dev/os/pull/26103 regressed buildability

As the cmake files reference public paths.... when they really should not.

--- a/clang-18.yaml
+++ b/clang-18.yaml
@@ -1,7 +1,7 @@
 package:
   name: clang-18
   version: 18.1.8
-  epoch: 1
+  epoch: 2
   description: "C language family frontend for LLVM"
   copyright:
     - license: Apache-2.0
@@ -70,7 +70,6 @@ pipeline:
         -DCLANG_BUILT_STANDALONE=TRUE \
         -DLLVM_COMMON_CMAKE_UTILS="/usr/lib/llvm-${{vars.major-version}}/share/cmake" \
         -DCLANG_CONFIG_FILE_SYSTEM_DIR=/etc/clang-${{vars.major-version}} \
-        -DLLVM_CONFIG=/usr/lib/llvm-${{vars.major-version}}/bin/llvm-config \
         -DCLANG_ENABLE_ARCMT=ON \
         -DLLVM_INCLUDE_TESTS=OFF \
         -DCLANG_ENABLE_STATIC_ANALYZER=ON \
@@ -80,7 +79,7 @@ pipeline:
         -DCLANG_VENDOR=Wolfi \
         -DENABLE_LINKER_BUILD_ID=ON \
         -DLIBCLANG_BUILD_STATIC=ON \
-        -DLLVM_INCLUDE_DIRS=/usr/lib/llvm-${{vars.major-version}}/include
+        -DCMAKE_PREFIX_PATH=/usr/lib/llvm-${{vars.major-version}}/
 
   - runs: |
       cmake --build build --target clang-tblgen

--- a/llvm-18.yaml
+++ b/llvm-18.yaml
@@ -1,7 +1,7 @@
 package:
   name: llvm-18
   version: 18.1.8
-  epoch: 2
+  epoch: 3
   description: "low-level virtual machine - core frameworks"
   copyright:
     - license: Apache-2.0
@@ -81,11 +81,9 @@ pipeline:
         ln -s ../lib/${{package.name}}/bin/$name "${{targets.contextdir}}"/usr/bin/$name
       done
 
-  - working-directory: ${{targets.contextdir}}/usr/lib/${{package.name}}
-    runs: |
+  - runs: |
       mkdir -p "${{targets.contextdir}}"/usr/lib/cmake
-      mv lib/cmake/llvm "${{targets.contextdir}}"/usr/lib/cmake/${{package.name}}
-      ln -s ${{package.name}} "${{targets.contextdir}}"/usr/lib/cmake/llvm
+      ln -s /usr/lib/"${{package.name}}"/lib/cmake/llvm "${{targets.contextdir}}"/usr/lib/cmake/llvm
 
   - uses: strip
 

